### PR TITLE
Add branch management

### PR DESCRIPTION
**Pull request by pierallard**
Add the new parameters from new version management of Crowdin.

The new API provides is_branch and branch parameters for add_file, add_directory and update_file.
https://crowdin.com/page/api

### CODE REVIEW

File: lib/Crowdin/Api/AddDirectory.php [hunk 1]
```diff
@@ -27,9 +31,16 @@ public function execute()
             $this->client->getProjectApiKey()
         );
 
-        $parameters = array_merge($this->parameters, array('name' => $this->getDirectory()));
+        $parameters = array_merge($this->parameters, ['name' => $this->getDirectory()]);
+        if ($this->getIsBranch()) {
+            $parameters['is_branch'] = '1';
+        }
+        if ($this->getBranch()) {
```
Comment (jjanvier) [inline lib/Crowdin/Api/AddDirectory.php]: null !==

File: lib/Crowdin/Api/AddFile.php [hunk 2]
```diff
@@ -43,6 +45,9 @@ public function execute()
         if ($this->getType()) {
             $data['type'] = $this->type;
         }
+        if ($this->getBranch()) {
```
Comment (jjanvier) [inline lib/Crowdin/Api/AddFile.php]: null !==

File: lib/Crowdin/Api/UpdateFile.php [hunk 3]
```diff
@@ -34,6 +37,9 @@ public function execute()
         );
 
         $data = $this->parameters;
+        if ($this->getBranch()) {
```
Comment (jjanvier) [inline lib/Crowdin/Api/UpdateFile.php]: null !==

**Comment by jjanvier** [comment]
I don't understand the getter/setter as you don't use them

**Comment by jjanvier** [comment]
gtm once fixed

File: lib/Crowdin/Api/AddDirectory.php [hunk 4]
```diff
@@ -51,4 +62,36 @@ public function getDirectory()
...
+    /**
+     * @return bool
+     */
+    public function getIsBranch()
+    {
+        return $this->isBranch;
+    }
+
+    /**
+     * @param $branch
```
Comment (skeleton) [inline lib/Crowdin/Api/AddDirectory.php]: phpdoc

File: lib/Crowdin/Api/AddDirectory.php [hunk 5]
```diff
@@ -51,4 +62,36 @@ public function getDirectory()
...
+     */
+    public function setBranch($branch)
+    {
+        $this->branch = $branch;
+    }
+
+    /**
+     * @param $branch
+     * @return mixed
```
Comment (skeleton) [inline lib/Crowdin/Api/AddDirectory.php]: same

File: lib/Crowdin/Api/UpdateFile.php [hunk 6]
```diff
@@ -77,4 +83,14 @@ public function getTranslations()
     {
         return $this->translations;
     }
+
+    public function getBranch()
+    {
+        return $this->branch;
+    }
+
+    public function setBranch($branch)
```
Comment (skeleton) [inline lib/Crowdin/Api/UpdateFile.php]: same

**Comment by skeleton** [comment]
gtm once fixed

**Pull request merged**